exercises/src/api/index.js

```diff
@@ -33,12 +33,19 @@ const start = function() {
     onFail: 'onError',
   });
 
+  connectModelUpdate(Exercise, 'deleteAttachment', {
+    method: 'DELETE',
+    pattern: 'exercises/{exerciseId}/attachments',
+    onSuccess: 'onAttachmentDelete',
+    onFail: 'onError',
+  });
+
 };
 
 const CSRF_TOKEN = get(document.head.querySelector('meta[name=csrf-token]'), 'content');
 
 Exercise.prototype.uploadImage = function(image, cb) {
-  const url = `/api/exercises/${this.number}@draft/attachments`;
+  const url = `/api/exercises/${this.uid}/attachments`;
```

Comment:
I think this might fail if the exercise is published?

Reply:
Yeah, I'd changed that in hopes that it'd fix the "needs draft to save" bug, and it did, but then the images don't show up for new drafts when it's loaded.  I didn't really mean to check this in, am glad you noticed